datafusion/src/row/mod.rs

```diff
@@ -349,6 +354,38 @@ mod tests {
                     assert_eq!(batch, output_batch);
                     Ok(())
                 }
+
+                #[test]
+                #[allow(non_snake_case)]
+                fn [<test_single_ $TYPE _nf>]() -> Result<()> {
```

Comment:
```suggestion
                fn [<test_single_ $TYPE _null_free>]() -> Result<()> {
```

Reply:
I was trying to remove unnecessary null bits checking staff when tuples are null_free according to their schema. Therefore, the null_free and not null_free code paths have diverged. 

To be specific, null bytes creation, checkings, settings will be removed for null-free tuples and their access methods.